File: OCH_Tao/ABC/B/260.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, X, Y, Z] = input[0].split(" ").map(x => parseInt(x, 10));
const A = input[1].split(" ").map(x => parseInt(x, 10));
const B = input[2].split(" ").map(x => parseInt(x, 10));
let C = [];
for (let i = 0; i < N; i++) {
  C.push({ "A": A[i], "B": B[i], "N": i + 1 });
}
let ans = [];
C.sort((a, b) => b.A - a.A || a.N - b.N);
for (const _ of Array(X)) {
  ans.push(C.shift().N);
}
C.sort((a, b) => b.B - a.B || a.N - b.N);
for (const _ of Array(Y)) {
  ans.push(C.shift().N);
}
C.sort((a, b) => (b.A + b.B) - (a.A + a.B) || a.N - b.N);
for (const _ of Array(Z)) {
  ans.push(C.shift().N);
}
console.log(ans.sort((a, b) => a - b).join("\n"));

File: OCH_Tao/ABC/B/260.py
N,X,Y,Z = map(int,input().split())
A = list(map(int,input().split()))
B = list(map(int,input().split()))
D = []
ans = set()
for i in range(N):
  D.append({"A":A[i],"B":B[i],"N":i+1})
D.sort(key=lambda x:(x["A"],-x["N"]))
for i in range(X):
  ans.add(D.pop()["N"])
D.sort(key=lambda x:(x["B"],-x["N"]))
for i in range(Y):
  ans.add(D.pop()["N"])
D.sort(key=lambda x:(x["A"]+x["B"],-x["N"]))
for i in range(Z):
  ans.add(D.pop()["N"])
print(*sorted(ans),sep="\n")